trinity/extensibility/plugin_manager.py

```diff
@@ -74,7 +113,55 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
             if not plugin.should_start():
                 continue
 
-            plugin.start(None)
+            plugin._start()
             self._started_plugins.append(plugin)
             self._logger.info("Plugin started: {}".format(plugin.name))
             self.broadcast(PluginStartedEvent(plugin), plugin)
+
+    def prepare(self,
+                args: Namespace,
+                chain_config: ChainConfig,
+                boot_kwargs: Dict[str, Any] = None) -> None:
+        for plugin in self._plugin_store:
+
+            if not self._is_responsible_for_plugin(plugin):
+                continue
+
+            context = self._create_context_for_plugin(plugin)
+            context.args = args
+            context.boot_kwargs = boot_kwargs
+            context.chain_config = chain_config
+            plugin.context = context
```

Comment:
Might be good to use a method for this `Plugin.set_context()` to be 1) explicit about the mutation and 2) allow the plugin to respond if needed.